front/src/components/boxs/chart/Chart.jsx

```diff
@@ -155,7 +161,15 @@ class Chartbox extends Component {
         emptySeries
       };
 
-      if (data.length > 0) {
+      // Before now, there was a "unit" attribute in this box instead of "units",
+      // so we need to support "unit" as some users may already have the box with that param
+      const unit = this.props.box.units ? this.props.box.units[0] : this.props.box.unit;
+      // We check if all deviceFeatures selected are in the same unit
+      const allUnitsAreSame = this.props.box.units ? allEqual(this.props.box.units) : false;
+
+      // If all deviceFeatures selected are in the same unit
+      // We do a average of all values
+      if (data.length > 0 && allUnitsAreSame) {
```

Comment:
Don't you prefer to check `data.length > 0` first, then check units?
A bit more efficient (just a bit).

Reply:
I don't see what you mean? Javascript do lazy-evaluation, so if data.length is false, it won't go to the next check

Reply:
But you still loop on "units" to check for single value, but for nothing.

Reply:
Ok I see! It's not a huge change, but at least i'll be more readable.

Fixed!